GitExtUtils/SubscribableTraceListener.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System.Diagnostics;
+using System.Text;
+
+namespace GitExtUtils;
+
+public delegate void TraceHandler(in string message);
+
+/// <summary>
+///   Provides an event for receiving Trace output.
+/// </summary>
+public interface ISubscribableTraceListener
+{
+    event TraceHandler TraceReceived;
+}
+
+public class SubscribableTraceListener : TraceListener, ISubscribableTraceListener
+{
+    private StringBuilder _trace = new();
+
+    public SubscribableTraceListener()
+    {
+        Trace.Listeners.Add(this);
+    }
+
+    protected override void Dispose(bool disposing)
+    {
+        if (disposing)
+        {
+            Trace.Listeners.Remove(this);
+        }
+
+        base.Dispose(disposing);
+    }
+
+    public event TraceHandler TraceReceived;
```

Comment:
[nit] move to the top of the class

Reply:
This is not a type but a provided event.
I followed https://learn.microsoft.com/en-us/archive/blogs/brada/internal-coding-guidelines#28-file-organization (Is it outdated?)
and took the liberty to put `Dispose` in the ctors section. Should I move `Dispose` to the methods section?

Reply:
I've largely been using that guidelines too (and pointing others at). The old event syntax (`public event EventHandler FooChanged { add { … } remove { … } }`) makes it look like properties, so in that context it makes sense. Though we haven't used that syntax in a long time (apart from dotnet/winforms codebase, hah). 
In this codebase (though we are not very consistent) many types define event handlers either before or right after .ctors. It generally makes it easier to discover.

But it's a nit.